### PR TITLE
Updates to devnet-sdk and the Operator Fee NAT Test to support live devnet testing

### PR DESCRIPTION
**Pull request by teddyknox**
This PR introduces several updates across the devnet SDK, Kurtosis deployer, and acceptance tests, and should fix CI:

**Devnet SDK & Kurtosis:**

*   **Added Node Naming:**
    *   Introduced a `Name` field to the `Node` descriptor (`devnet-sdk/descriptors/deployment.go`) and the `Node` interface/implementation (`devnet-sdk/system/interfaces.go`, `devnet-sdk/system/node.go`).
    *   The node name is now passed during node creation (`devnet-sdk/system/chain.go`, `devnet-sdk/system/node.go`) and utilized in tests for node filtering (e.g., `devnet-sdk/testing/systest/multi_client.go`).
    *   Added `Scheme` to `PortInfo` descriptor (`devnet-sdk/descriptors/deployment.go`) and improved endpoint handling in `devnet-sdk/system/chain.go`.
    *   Implemented the `Name()` method in various mock nodes for testing (`devnet-sdk/system/txbuilder_test.go`, `devnet-sdk/testing/testlib/validators/validators_test.go`, `op-acceptance-tests/tests/interop/mocks_test.go`).
*   **Propagated L1 Chain Configuration:**
    *   Added `L1ChainConfig` field to `DeployerData` (`kurtosis-devnet/pkg/kurtosis/sources/deployer/deployer.go`).
    *   Modified the deployer (`kurtosis-devnet/pkg/kurtosis/sources/deployer/wallets.go`) to parse and return the full `params.ChainConfig` from the L1 genesis file, renaming `getL1ChainID` to `getConfig`.
    *   Passed the `L1ChainConfig` through Kurtosis (`kurtosis-devnet/pkg/kurtosis/kurtosis.go`) to the `devnet-sdk` (`devnet-sdk/system/chain.go`).
    *   Removed a placeholder error message in `chain.Config()` (`devnet-sdk/system/chain.go`), allowing L1 chains to access their configuration.

**Acceptance Tests (Operator Fee):**

*   **Improved Test Case Generation (`op-acceptance-tests/tests/isthmus/operator_fee/test_case_generator.go`):**
    *   Replaced deprecated `rand.Seed` with modern `rand.New(rand.NewSource(...))` and added `crypto/rand` usage for generating `big.Int` values.
    *   Adjusted specific hardcoded values used for testing edge cases.
*   **Increased Sync Delay (`op-acceptance-tests/tests/isthmus/operator_fee/operator_fee_test.go`):**
    *   Increased the waiting time after setting operator fees from 30 seconds to 2 minutes to allow L2 nodes sufficient time to sync.

### CODE REVIEW

File: devnet-sdk/system/chain.go [hunk 1]
```diff
@@ -214,6 +219,7 @@ func newL2ChainFromDescriptor(d *descriptors.L2Chain) (*l2Chain, error) {
 	// TODO: handle incorrect descriptors better. We could panic here.
 
 	nodes := newNodesFromDescriptor(&d.Chain)
+	// print a message with the content of d.Config
```
Comment (scharissis) [inline devnet-sdk/system/chain.go]: This seems out of place?

File: devnet-sdk/testing/systest/multi_client.go [hunk 2]
```diff
@@ -30,10 +31,13 @@ func getEthClients(chain system.Chain) ([]HeaderProvider, error) {
 	hps := make([]HeaderProvider, 0, len(chain.Nodes()))
 	for _, n := range chain.Nodes() {
 		gethCl, err := n.GethClient()
+
 		if err != nil {
 			return nil, fmt.Errorf("failed to get geth client: %w", err)
 		}
-		hps = append(hps, gethCl)
+		if !regexp.MustCompile(`snapsync-\d+$`).MatchString(n.Name()) {
```
Comment (scharissis) [inline devnet-sdk/testing/systest/multi_client.go:38]: Do we need to maintain such a blacklist of names/regexes?

**Review by scharissis — APPROVED** [review]
Nice, thanks!
I left a few comments on small things, but LGTM.

File: devnet-sdk/testing/systest/multi_client.go (hunk 2 above)
Comment (teddyknox) [reply devnet-sdk/testing/systest/multi_client.go:38]: We do, snapsync nodes ironically do not serve RPC methods correctly.

File: devnet-sdk/system/chain.go (hunk 1 above)
Comment (teddyknox) [reply devnet-sdk/system/chain.go]: Removed.
